src/job_handler_plugins/local_container/__init__.py

```diff
@@ -58,7 +58,7 @@ def start(self) -> str:
 
     def remove(self) -> Tuple[str, str]:
         try:
-            container = self.client.containers.get(self.job.entity["name"])
+            container = self.client.containers.get(self.job.entity.get("name", "local-job"))
```

Comment:
If the containers name is generated, you will need to set that in the entity when starting the job. And if it's not there. remove() must fail.